lib/galaxy/webapps/galaxy/api/visualizations.py

```diff
@@ -239,175 +231,47 @@ def set_slug(
 
+    @router.get(
+        "/api/visualizations/{id}",
+        summary="Get a visualization by ID.",
+    )
+    def show(
+        self,
+        id: VisualizationIdPathParam,
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> VisualizationShowResponse:
+        """Return the visualization."""
+        return self.service.show(trans, id)
 
-class VisualizationsController(BaseGalaxyAPIController, UsesVisualizationMixin, UsesAnnotations):
-    """
-    RESTful controller for interactions with visualizations.
-    """
-
-    service: VisualizationsService = depends(VisualizationsService)
-
-    @expose_api
-    def show(self, trans: GalaxyWebTransaction, id: str, **kwargs):
-        """
-        GET /api/visualizations/{viz_id}
-        """
-        # TODO: revisions should be a contents/nested controller like viz/xxx/r/xxx)?
-        # the important thing is the config
-        # TODO:?? /api/visualizations/registry -> json of registry.listings?
-        visualization = self.get_visualization(trans, id, check_ownership=False, check_accessible=True)
-        dictionary = trans.security.encode_dict_ids(self.get_visualization_dict(visualization))
-        dictionary["url"] = web.url_for(
-            controller="visualization",
-            action="display_by_username_and_slug",
-            username=visualization.user.username,
-            slug=visualization.slug,
-        )
-        dictionary["username"] = visualization.user.username
-        dictionary["email_hash"] = md5_hash_str(visualization.user.email)
-        dictionary["tags"] = visualization.make_tag_string_list()
-        dictionary["annotation"] = self.get_item_annotation_str(trans.sa_session, trans.user, visualization)
-        # need to encode ids in revisions as well
-        encoded_revisions = []
-        for revision in dictionary["revisions"]:
-            # NOTE: does not encode ids inside the configs
-            encoded_revisions.append(trans.security.encode_id(revision))
-        dictionary["revisions"] = encoded_revisions
-        dictionary["latest_revision"] = trans.security.encode_dict_ids(dictionary["latest_revision"])
-        if trans.app.visualizations_registry:
-            visualization = trans.app.visualizations_registry.get_plugin(dictionary["type"])
-            dictionary["plugin"] = visualization.to_dict()
-        return dictionary
-
-    @expose_api
-    def create(self, trans: GalaxyWebTransaction, payload: dict, **kwargs):
+    @router.post(
+        "/api/visualizations",
+        summary="Create a new visualization.",
+    )
+    def create(
+        self,
+        payload: VisualizationCreatePayload = Body(...),
+        import_id: Optional[DecodedDatabaseIdField] = Query(
+            None, title="Import ID", description="The encoded database identifier of the Visualization to import."
+        ),
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> VisualizationCreateResponse:
         """
         POST /api/visualizations
```

Comment:
The dosctring ends up as documentation in the schema, we don't need to include stuff like `POST /api/visualizations` there.